src/components/modals/FlipInputModal2.tsx

```diff
@@ -0,0 +1,301 @@
+        <ExchangeRate2 wallet={wallet} tokenId={tokenId} style={styles.rateBalanceText} />
+      </View>
+    )
+  })
+
+  const renderBalance = useHandler(() => {
+    const { multiplier, name } = displayDenom
+    const balanceCrypto = balances[currencyCode]
+    const balance = `${formatNumber(div(balanceCrypto, multiplier, DECIMAL_PRECISION))} ${name} `
+    return (
+      <View style={styles.rateBalanceContainer}>
+        <EdgeText style={styles.secondaryTitle}>{s.strings.send_confirmation_balance}</EdgeText>
+        <EdgeText style={styles.rateBalanceText}>
+          {balance}
+          (
+          <FiatText wallet={wallet} tokenId={tokenId} nativeCryptoAmount={balanceCrypto} />)
+        </EdgeText>
+      </View>
+    )
+  })
+
+  const renderFees = useHandler(() => {
+    const feeDisplayAmountTemp = div(feeNativeAmount, feeDisplayDenom.multiplier, DECIMAL_PRECISION)
+    const feeDisplayAmount = toFixed(feeDisplayAmountTemp, 0, log10(feeDisplayDenom.multiplier))
+
+    const feeCryptoText = `${feeDisplayAmount} ${feeDisplayDenom.name} `
+    const feeTextStyle = feeStyle === 'dangerText' ? styles.feeTextDanger : feeStyle === 'warningText' ? styles.feeTextWarning : styles.feeTextDefault
+
+    return (
+      <View style={styles.feeContainer}>
+        <View style={styles.feeTitleContainer}>
+          <EdgeText style={styles.primaryTitle}>{s.strings.string_fee}</EdgeText>
+          {onFeesChange ? <FontAwesomeIcon name="edit" style={styles.feeIcon} size={theme.rem(0.75)} /> : null}
+        </View>
+        <EdgeText style={feeTextStyle}>
+          {feeCryptoText}
+          (<FiatText nativeCryptoAmount={feeNativeAmount} wallet={wallet} />)
+        </EdgeText>
+      </View>
+    )
+  })
+
+  const renderFlipInput = useHandler(() => {
+    return (
+      <Card>
+        <ExchangedFlipInput2
+          ref={exchangedFlipInputRef}
+          walletId={walletId}
+          tokenId={tokenId}
+          startNativeAmount={startNativeAmount}
+          forceField={amounts.fieldChanged}
+          headerText={flipInputHeaderText}
+          onAmountChanged={handleAmountsChanged}
+          keyboardVisible
+          onNext={handleCloseModal}
+        />
+        {getSpecialCurrencyInfo(pluginId).noMaxSpend !== true && hideMaxButton !== true ? (
+          <MiniButton alignSelf="center" label={s.strings.string_max_cap} marginRem={[1.2, 0, 0]} onPress={handleSendMaxAmount} />
+        ) : null}
+      </Card>
+    )
+  })
+
+  React.useImperativeHandle(ref, () => ({
+    setFees: (params: SetFeesParams) => {
+      setFeeTokenId(params.feeTokenId)
+      setFeeNativeAmount(params.feeNativeAmount)
+    },
+    setAmount: (field: ExchangeFlipInputFields, value: string) => exchangedFlipInputRef.current?.setAmount(field, value),
+    setError: (errorMessage: string | null) => setErrorMessage(errorMessage)
+  }))
+
+  return (
+    <ThemedModal bridge={bridge} onCancel={handleCloseModal}>
```

Comment:
`onCancel` should resolve to `undefined`, not an amount.

Reply:
Again, retained from FlipInputModal but I see why. The modal is actually updating the parent scene on every keystroke, so even if the user cancels, the parent was already being updated. Basically the modal has no cancel functionality since so much happens while the user is typing. ie. recreating the tx, makeSpend, etc. It would be hard to "unwind" all the user input.

Reply:
I see. This seems like a bug in the send scene, which we are now paying for in the modal. I was hoping the send scene refactor would clean up stuff like this.

Reply:
I think it's actually better to do all the tx calculation on the parent scene. To do it in the modal would require the modal to be "smart" which makes it unusable in other scenes that aren't simply doing spends. This was the cause of a lot of hacky redux code. It's not impossible to unwind but quite difficult and I don't think the result would be much better. I think of the FlipInputModal as somewhat like the FlipInput itself.. it's mostly dumb, and the parent controls it. It just happens to be in a modal vs a component and it has more UI to control.